src/settings/NumberInput.tsx

```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+
+type NumberInputProps = {
+  label: string;
+  value: number;
+  onChange: (x: number) => void;
+};
+
+export default function NumberInput({
+  label,
+  value,
+  onChange,
+}: NumberInputProps) {
+  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const { value } = e.target;
+    if (Number.isSafeInteger(Number(value))) {
+      onChange(Number(value));
+    }
+  };
+
+  return (
+    <React.Fragment>
+      <input
+        type='text'
+        id={label}
+        name={label}
+        value={value}
+        onChange={handleChange}
+      />
+      <label htmlFor={label}>{label}</label>
```

Comment:
I do not think that label should be an id.

Reply:
Yes, I should probably nest input in label, bcs I didn't have an id.